openml/config.py

```diff
@@ -255,6 +259,7 @@ def _setup(config: _Config | None = None) -> None:
     avoid_duplicate_runs = config["avoid_duplicate_runs"]
     apikey = config["apikey"]
     server = config["server"]
+    show_progress = config.get("show_progress", _defaults["show_progress"])
```

Comment:
Should this explicitly cast to bool somewhere? I found this line in the parsing of the Config which seems to suggest it needs explicit handling to convert to booleans.

https://github.com/openml/openml-python/blob/8d0324374733795ca05975b2427757c12e24d949/openml/config.py#L336-L340

Reply:
Yikes, can't believe I missed that. Thanks. 
It turns out that this didn't work either. Setting avoid_duplicate_runs would actually give me a TypeError (wasn't allowed to set a boolean value to the Config object), so I rewrote this and added a regression test.